protocol-units/bridge/contracts/src/AtomicBridgeCounterparty.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity ^0.8.22;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {IAtomicBridgeCounterparty} from "./IAtomicBridgeCounterparty.sol";
+import {AtomicBridgeInitiator} from "./AtomicBridgeInitiator.sol";
+
+contract AtomicBridgeCounterparty is IAtomicBridgeCounterparty, OwnableUpgradeable {
+    enum MessageState {
+        PENDING,
+        COMPLETED,
+        REFUNDED
+    }
+
+    struct BridgeTransferDetails {
+        bytes32 initiator;  
+        address recipient;
+        uint256 amount;
+        bytes32 hashLock;
+        uint256 timeLock; 
+        MessageState state; 
+    }
+
+    AtomicBridgeInitiator public atomicBridgeInitiator;
+    mapping(bytes32 => BridgeTransferDetails) public bridgeTransfers; 
+
+    function initialize(address _atomicBridgeInitiator, address owner) public initializer {
+        if (_atomicBridgeInitiator == address(0)) revert ZeroAddress();
+        atomicBridgeInitiator = AtomicBridgeInitiator(_atomicBridgeInitiator);
+        __Ownable_init(owner);
+    }
+
+    function setAtomicBridgeInitiator(address _atomicBridgeInitiator) external onlyOwner {
+        if (_atomicBridgeInitiator == address(0)) revert ZeroAddress();
+        atomicBridgeInitiator = AtomicBridgeInitiator(_atomicBridgeInitiator);
+    }
+
+    function lockBridgeTransferAssets(
+        bytes32 initiator,
+        bytes32 bridgeTransferId,
+        bytes32 hashLock,
+        uint256 timeLock,
+        address recipient,
+        uint256 amount
+    ) external onlyOwner returns (bool) {
+        if (amount == 0) revert ZeroAmount();
+        if (atomicBridgeInitiator.poolBalance() < amount) revert InsufficientWethBalance();
+
+        bridgeTransfers[bridgeTransferId] = BridgeTransferDetails({
+            recipient: recipient,
+            initiator: initiator,
+            amount: amount,
+            hashLock: hashLock,
+            timeLock: block.number + timeLock, // using block number for timelock
+            state: MessageState.PENDING 
+        });
+
+        emit BridgeTransferAssetsLocked(bridgeTransferId, recipient, amount, hashLock, timeLock);
+        return true;
+    }
+
+    function completeBridgeTransfer(bytes32 bridgeTransferId, bytes32 preImage) external {
+        BridgeTransferDetails storage details = bridgeTransfers[bridgeTransferId];
+        if (details.state != MessageState.PENDING) revert BridgeTransferStateNotPending();
+        bytes32 computedHash = keccak256(abi.encodePacked(preImage));
+        if (computedHash != details.hashLock) revert InvalidSecret();
+        if (block.number > details.timeLock) revert TimeLockNotExpired();
+
+        details.state = MessageState.COMPLETED;
+
+        atomicBridgeInitiator.withdrawWETH(details.recipient, details.amount);
+
+        emit BridgeTransferCompleted(bridgeTransferId, preImage);
+    }
+
+    function abortBridgeTransfer(bytes32 bridgeTransferId) external {
```

Comment:
Why is it allowed for anyone to call this? Are we claiming security by obscurity? The `bridgeTransferId` here is not being used as a secret to compute a different identifier internally.

Reply:
Should be restricted to owner. Will update.